### ISSUE TITLE
Enforce `x`, `x1`,` x2` naming consistency 

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1381]
# Rule request

## Thesis

I often see that some people use this naming pattern:

```python
x1 = 1
x2 = 2
```

Which is perfectly fine.
But, this one looks ugly:

```
x = 1
x2 = 2
```

Note `x1` vs `x2` difference.

## Reasoning

My proposed solution is to create a new naming rule to check that if a variable named `something1` or `something2` there should not be just `something` in the current scope. While just `something` and `something1` with `something2` are totally fine.

Our own code will have lots of these issues!

**Comment by sobolevn** [comment]
`x0` is always fine.

**Comment by raghul-rajasekar** [comment]
Hi, I'd like to claim this issue.

**Comment by sobolevn** [comment]
Thanks a lot, @raghul-rajasekar!

**Comment by toennifer** [comment]
Hi, I am a Umich student and would like to work on this issue for one of my class assignments!

I have a question regarding the rule:
Would just having something like `x2 = 2` (without `x1 = 1`) count as violating the rule?

**Comment by sobolevn** [comment]
No, any names with a context are fine. We only look for this pattern:

```
x = 1
x2 = 2
```

**Comment by sobolevn** [comment]
Thanks a lot, assigned!

**Comment by Dreamsorcerer** [comment]
Might also want to consider checking for underscores:
```
x = 1
x_2 = 2
```

**Comment by sobolevn** [comment]
I am wondering: what context should we care about? One function? Or the whole module?

**Comment by sobolevn** [comment]
Ok, I am going to only check for names in the same context. Whatever it might be: function, class, module.

**Comment by sobolevn** [comment]
Ok, this is hard. There are lots of corner cases and small things to think about.

For example, attributes need special handling:
- `some.x` and `some.x2` are violating this rule
- while `some.x` and `other.x2` are not

So, I am moving this feature for the next version.
